src/test/java/mate/academy/dao/UserDaoTest.java

```diff
@@ -0,0 +1,123 @@
+package mate.academy.dao;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.dao.impl.RoleDaoImpl;
+import mate.academy.dao.impl.UserDaoImpl;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class UserDaoTest extends AbstractTest {
+    private static final String EMAIL = "tom@gmail.com";
+    private static final String PASSWORD = "1234";
+    private UserDao userDao;
+    private RoleDao roleDao;
+    private Role adminRole;
+    private Role userRole;
+    private User actualUser;
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[]{User.class, Role.class};
+    }
+
+    @BeforeEach
+    void setUp() {
+        userDao = new UserDaoImpl(getSessionFactory());
+        roleDao = new RoleDaoImpl(getSessionFactory());
+        adminRole = roleDao.save(new Role(Role.RoleName.ADMIN));
+        userRole = roleDao.save(new Role(Role.RoleName.USER));
+    }
+
+    @Test
+    void save_Ok() {
+        actualUser = userDao.save(getUserForTest());
+        Assertions.assertNotNull(actualUser);
+        Assertions.assertEquals(1L, actualUser.getId());
+    }
+
+    @Test
+    void findByEmail_Ok() {
+        actualUser = userDao.save(getUserForTest());
+        Optional<User> userByEmail = userDao.findByEmail(EMAIL);
+        Assertions.assertTrue(userByEmail.isPresent());
+        User userFindByEmail = userByEmail.get();
+        Assertions.assertNotNull(userFindByEmail);
+        Assertions.assertEquals(userFindByEmail.getEmail(), actualUser.getEmail());
+    }
+
+    @Test
+    void findById_Ok() {
+        actualUser = userDao.save(getUserForTest());
+        User userId = userDao.findById(1L).orElseThrow();
+        Assertions.assertNotNull(userId);
+        Assertions.assertEquals(userId.getId(), actualUser.getId());
+    }
+
+    @Test
+    void findByEmail_WrongEmail_NotOk() {
+        userDao.save(getUserForTest());
+        Optional<User> userByEmail = userDao.findByEmail("invalid_email");
+        Assertions.assertTrue(userByEmail.isEmpty());
+    }
+
+    @Test
+    void findByEmail_NotFound_NotOk() {
+        try {
```

Comment:
This test case seems to be incorrect. You are expecting a DataProcessingException when the email is not found, but the findByEmail method returns an Optional, which should be empty if the email is not found. Remove this test case or modify it to check for an empty Optional.